tests/test_obscore.py

```diff
@@ -0,0 +1,468 @@
+        self.assertEqual(table_spec.fields["em_xel"].dtype, sqlalchemy.Integer)
+
+    def test_insert_existing_collection(self):
+        """Test insert and import registry methods, with various restrictions
+        on collection names.
+        """
+
+        # First item is collections, second item is expected record count.
+        test_data = (
+            (None, 6),
+            (["run1", "run2"], 2),
+            (["run[34]"], 2),
+            (["[rR]un[^6]"], 5),
+        )
+
+        for collections, count in test_data:
+            for do_import in (False, True):
+
+                registry = self.make_registry(collections)
+                self._insert_datasets(registry, do_import)
+
+                rows = self._obscore_select(registry)
+                self.assertEqual(len(rows), count)
+
+    def test_drop_datasets(self):
+        """Test for dropping datasets after obscore insert."""
+
+        collections = None
+        registry = self.make_registry(collections)
+        refs = self._insert_datasets(registry)
+
+        rows = self._obscore_select(registry)
+        self.assertEqual(len(rows), 6)
+
+        # drop single dataset
+        registry.removeDatasets(ref for ref in refs if ref.run == "run1")
+        rows = self._obscore_select(registry)
+        self.assertEqual(len(rows), 5)
+
+        # drop whole run collection
+        registry.removeCollection("run6")
+        rows = self._obscore_select(registry)
+        self.assertEqual(len(rows), 4)
+
+    def test_associate(self):
+        """Test for associating datasets to TAGGED collection."""
+
+        collections = ["tagged"]
+        registry = self.make_registry(collections, "TAGGED")
+        refs = self._insert_datasets(registry)
+
+        rows = self._obscore_select(registry)
+        self.assertEqual(len(rows), 0)
+
+        # Associate datasets that are already in obscore, changes nothing.
+        registry.associate("tagged", (ref for ref in refs if ref.run == "run1"))
+        rows = self._obscore_select(registry)
+        self.assertEqual(len(rows), 1)
+
+        # Associate datasets that are not in obscore
+        registry.associate("tagged", (ref for ref in refs if ref.run == "run3"))
+        rows = self._obscore_select(registry)
+        self.assertEqual(len(rows), 2)
+
+        # Disassociate them
+        registry.disassociate("tagged", (ref for ref in refs if ref.run == "run3"))
+        rows = self._obscore_select(registry)
+        self.assertEqual(len(rows), 1)
+
+        # Non-associated dataset, should be OK and not throw.
+        registry.disassociate("tagged", (ref for ref in refs if ref.run == "run2"))
+        rows = self._obscore_select(registry)
+        self.assertEqual(len(rows), 1)
+
+        registry.disassociate("tagged", (ref for ref in refs if ref.run == "run1"))
+        rows = self._obscore_select(registry)
+        self.assertEqual(len(rows), 0)
+
+
+class SQliteObsCoreTest(ObsCoreTests, unittest.TestCase):
```

Comment:
```suggestion
class SqliteObsCoreTest(ObsCoreTests, unittest.TestCase):
```
or `SQLite`, if you prefer.  Just not `SQlite` :slightly_smiling_face: .